tests/test_models/test_engine/test_db_storage.py

```diff
@@ -68,21 +69,113 @@ def test_dbs_func_docstrings(self):
+        new = BaseModel()
+        new.save()
+        temp = storage.all()
+        self.assertIsInstance(temp, dict)
+        dict_key = "{}.{}".format("BaseModel", new.id)
+        self.assertIn(dict_key, temp)
+
+        # tests storage.all() with cls argument, with no cls instances
+        all_states = storage.all(State)
+        db = MySQLdb.connect(host="localhost", user="hbnb_test",
+                             passwd="hbnb_test_pd", db="hbnb_test_db")
+        cur = db.cursor()
+        cur.execute("""SELECT COUNT(*) FROM states""")
+        count = cur.fetchall()
+        self.assertEqual(len(all_states.keys()), count)
+        # creates cls instance and retests storage.all() with cls
+        new_state = State()
+        new_state.name = "California"
+        new_state.save()
+        all_states = storage.all(State)
+        self.assertEqual(len(all_states.keys()), count + 1)
+        for k, v in all_states.items():
+            self.assertEqual("California", v.name)
+        cur.execute("""SELECT COUNT(*) FROM states""")
+        count2 = cur.fetchall()
+        self.assertEqual(count + 1, count2)
+        # tests delete method
+        storage.delete(new_state)
+        all_states = storage.all(State)
+        self.assertEqual(len(all_states.keys()), count)
+        cur.close()
+        db.close()
+
+    @skipIf(storage_t != 'db', "not testing db storage")
+    def test_empty(self):
+        """ Data is saved to database """
+        new = State()
+        new.name = "California"
+        new.save()
+        C_id = new.id
+        db = MySQLdb.connect(host="localhost", user="hbnb_test",
+                             passwd="hbnb_test_pd", db="hbnb_test_db")
+        cur = db.cursor()
+        cur.execute("""SELECT COUNT(*) FROM states""")
+        count = cur.fetchall()
+        self.assertNotEqual(count, 0)
+        cur.close()
+        db.close()
+
+    @skipIf(storage_t != 'db', "not testing db storage")
+    def test_reload(self):
+        """ Storage database is successfully loaded to __objects """
+        new = State()
+        new.name = "testing"
+        new.save()
+        storage.reload()
+        self.assertIn("testing", storage.all(State).values())
+
+    @skipIf(storage_t != 'db', "not testing db storage")
+    def test_type_objects(self):
+        """ Confirm __objects is a dict """
+        self.assertEqual(type(storage.all()), dict)
+
+    @skipIf(storage_t != 'db', "not testing db storage")
+    def test_key_format(self):
+        """ Key is properly formatted """
+        new = BaseModel()
+        new.save()
+        _id = new.id
+        for key in storage.all().keys():
+            temp = key
+            if _id in key:
+                self.assertEqual(temp, 'BaseModel' + '.' + _id)
+
+    @skipIf(storage_t != 'db', "not testing db storage")
+    def test_storage_var_created(self):
+        """ DBStorage object storage created """
+        from models.engine.db_storage import DBStorage
```

Comment:
Isn't this already imported at the top?

Reply:
Already fixed per our previous pair programming call :)